README.md

```diff
@@ -19,8 +19,240 @@
 </p>
 
 
-With Kairos you can build immutable, bootable Kubernetes and OS images for your edge devices as easily as writing a Dockerfile. Optional P2P mesh with distributed ledger automates node bootstrapping and coordination. Updating nodes is as easy as CI/CD: push a new image to your container registry and let secure, risk-free A/B atomic upgrades do the rest.
+## What is immucore?
 
+---
+
+Immucore is the management interface to mount Kairos disks and filesystems.
+Is a dracut module responsible for mounting the root tree during boot time with the immutable specific setup.
+The immutability concept refers to read only root (/) system.
+To ensure the linux OS is still functional certain paths or areas are required to be writable,
+in those cases an ephemeral overlay tmpfs is set in place.
+
+Additionally, the immutable rootfs module can also mount a custom list of device blocks with read write permissions, those are mostly devoted to store persistent data.
+
+
+Immucore is mostly configured via kernel command line parameters or via the `/run/cos/cos-layout.env` environment file.
+
+These are the read write paths the module mounts as part of the overlay
+ephemeral tmpfs: `/etc`, `/root`, `/home`, `/opt`, `/srv`, `/usr/local`
+and `/var`.
+
+
+## Kernel configuration parameters
+
+---
+
+The immutable rootfs can be configured with the following kernel parameters:
+
+* `cos-img/filename=<imgfile>`: This is one of the main parameters, it defines
+  the location of the image file to boot from. This defines the booting mode for
+  Immucore, setting in motion the full DAG to set up the system.
```

Comment:
Jargon - DAG is first mentioned here. The naive user does not know what this means.
Can this be replaced?